### ISSUE TITLE
forceGlobalBucketAccessEnabled does not work when dualstackEnabled is set for S3

### ISSUE THREAD

**Issue by kwadhwa18** [aws/aws-sdk-java#2799]
### Describe the bug

I'm trying to use [dual stack for S3 ](https://docs.aws.amazon.com/AmazonS3/latest/userguide/dual-stack-endpoints.html). Here is how I'm constructing the client:

```
    s3 =
        AmazonS3ClientBuilder.standard()
            .withClientConfiguration(configuration)
            .withCredentials(credentialsProvider)
            .withForceGlobalBucketAccessEnabled(true)
            .withDualstackEnabled(true)
            .build();
```

accessing a bucket with this client fails with the following error message:

```
 The bucket is in this region: us-east-2. Please use this region to retry the request (Service: Amazon S3; Status Code: 301; Error Code: PermanentRedirect; Request ID: .....; S3 Extended Request ID: .....
```

Although, if I don't set dual stack like below, accessing a bucket in any region is successful:

```
    s3 =
        AmazonS3ClientBuilder.standard()
            .withClientConfiguration(configuration)
            .withCredentials(credentialsProvider)
            .withForceGlobalBucketAccessEnabled(true)
            .build();
```

### Expected Behavior

The client should be able to access bucket/object in any region with forceGlobalBucketAccessEnabled set to true (even when dual stack is set)

### Current Behavior

operation fails with error code 301

### Reproduction Steps

Mentioned above in the description

### AWS Java SDK version used

1.12.130

### JDK version used

java version "17.0.2" 2022-01-18 LTS

### Operating System and version

macOS Monterey 12.0.1

**Comment by yasminetalby** [comment]
Hello @kwadhwa18 ,

Thank you very much for your submission. I was able to reproduce this issue and confirm that it is a bug.
We will be adding this bug fix to our backlog, however as we are currently focusing our effort on SDK v2, I would advise to not use forceGlobalBucketAccessEnabled with dual stack.

Best,

Yasmine

**Comment by kwadhwa18** [comment]
thanks @yasminetalby

**Comment by islamismailov** [comment]
This issue still persists on the latest version of v1 SDK. This SDK is still widely used by a lot of software. Can we prioritize the fix please? @yasminetalby

**Comment by debora-ito** [comment]
A fix was released in SDK version `1.12.440`, please check it out.
